packages/client/modules/userDashboard/components/OrgIntegrations/UpsertGitLabProviderDialog.tsx

```diff
@@ -0,0 +1,151 @@
+import useForm from '../../../../hooks/useForm'
+import {Dialog} from '../../../../ui/Dialog/Dialog'
+import {DialogContent} from '../../../../ui/Dialog/DialogContent'
+import {DialogTitle} from '../../../../ui/Dialog/DialogTitle'
+import linkify from '../../../../utils/linkify'
+import makeAppURL from '../../../../utils/makeAppURL'
+import Legitity from '../../../../validation/Legitity'
+import CopyShortLink from '../../../meeting/components/CopyShortLink/CopyShortLink'
+
+type Props = {
+  isOpen: boolean
+  onClose: () => void
+  defaults: {
+    serverBaseUrl: string
+    clientId: string
+    clientSecret: string
+  }
+  onUpsert: (serverBaseUrl: string, clientId: string, clientSecret: string) => void
+  error?: {message: string}
+}
+
+const UpsertGitLabProviderDialog = (props: Props) => {
+  const {isOpen, onClose, defaults, onUpsert, error} = props
+
+  const redirectUri = makeAppURL(window.location.origin, 'auth/gitlab')
+  const {fields, onChange, validateField} = useForm({
+    serverBaseUrl: {
+      getDefault: () => defaults.serverBaseUrl,
+      validate: (rawInput: string) => {
+        return new Legitity(rawInput).test((maybeUrl) => {
+          if (!maybeUrl) return 'Please enter a server base URL'
+          const links = linkify.match(maybeUrl)
+          return !links ? 'Not looking too linky' : ''
+        })
+      }
+    },
+    clientId: {
+      getDefault: () => defaults.clientId,
+      validate: (clientId: string) => {
+        return new Legitity(clientId).required('Please enter a client ID')
+      }
+    },
+    clientSecret: {
+      getDefault: () => defaults.clientSecret,
+      validate: (clientSecret: string) => {
+        return new Legitity(clientSecret).required('Please enter a client secret')
+      }
+    }
+  })
+
+  const onSubmit = () => {
+    const validation = validateField()
+    if (Object.values(validation).some(({error}) => error)) return
+    onUpsert(fields.serverBaseUrl.value, fields.clientId.value, fields.clientSecret.value)
+    onClose?.()
+  }
+
+  return (
+    <Dialog isOpen={isOpen} onClose={onClose}>
+      <DialogContent>
+        <DialogTitle className='flex items-center'>
+          <GitLabProviderLogo />
+          <div className='ml-2'>Add GitLab Server</div>
+        </DialogTitle>
+        <form
+          onSubmit={(e) => {
+            e.preventDefault()
+            onSubmit()
+          }}
+        >
+          <div className='pt-6 pb-2'>
+            In the <b>Admin Area</b>, <b>Applications</b> add a <b>New application</b> with the
+            following settings
+          </div>
+          <div className='flex items-center p-2'>
+            <div className='w-36'>Name</div>
+            <div>Parabol</div>
+          </div>
+          <div className='flex items-center p-2'>
+            <div className='w-36'>Redirect URI</div>
+            <CopyShortLink url={redirectUri} title='Redirect URL' tooltip='Copy Redirect URL' />
```

Comment:
+1 this looked like a hyperlink to me. I tried to copy the text by highlighting it, but I wasn't able to highlight it. I then accidentally clicked on it and saw the "Copy Redirect URL" tooltip. 

I think it'd be more user friendly if it had a copy button next to the URL, or if it looked more similar to the invite link:

![Screenshot 2024-09-09 at 19 06 51](https://github.com/user-attachments/assets/c22a5e9d-379c-4338-bc13-837395b03ec4)


![Screenshot 2024-09-09 at 19 07 36](https://github.com/user-attachments/assets/d76d690e-d552-4615-ba9f-6f80a06366f2)

Reply:
<img width="722" alt="image" src="https://github.com/user-attachments/assets/661efe38-5dc3-4f16-98a1-0f9e538aa1d9">

Reply:
Love it!